internal/cloud/metadata.go

```diff
@@ -83,24 +89,24 @@ func identifyGcp() (bool, error) {
 	return regexp.MatchString(".*Google.*", provider)
 }
 
-func IdentifyCloudProvider() (string, error) {
+func (c *Identifier) IdentifyCloudProvider() (string, error) {
 	log.Info("Identifying if the VM is running in a cloud environment...")
 
-	if result, err := identifyAzure(); err != nil {
+	if result, err := c.identifyAzure(); err != nil {
 		return "", err
 	} else if result {
 		log.Infof("VM is running on %s", Azure)
 		return Azure, nil
 	}
 
-	if result, err := identifyAws(); err != nil {
+	if result, err := c.identifyAws(); err != nil {
```

Comment:
This is probably out of scope for this PR so feel free to ignore, but should it be `identifyAWS` instead?